flink-runtime/src/main/java/org/apache/flink/streaming/runtime/operators/sink/committables/CheckpointCommittableManagerImpl.java

```diff
@@ -147,15 +146,16 @@ Collection<CommittableWithLineage<CommT>> drainFinished() {
     }
 
     CheckpointCommittableManagerImpl<CommT> merge(CheckpointCommittableManagerImpl<CommT> other) {
-        checkArgument(Objects.equals(other.checkpointId, checkpointId));
+        checkArgument(other.checkpointId == checkpointId);
```

Comment:
Nit: This change should probably go into the commit, changing the type of the checkpoint fields

Reply:
Good catch. I'll try to move it.